rllib/offline/offline_prelearner.py

```diff
@@ -240,17 +267,39 @@ def _map_to_episodes(
                 episode = SingleAgentEpisode(
                     id_=batch[schema[Columns.EPS_ID]][i],
                     agent_id=agent_id,
+                    # Observations might be (a) serialized and/or (b) converted
+                    # to a JSONable (when a composite space was used). We unserialize
+                    # and then reconvert from JSONable to space sample.
                     observations=[
-                        unpack_if_needed(obs),
-                        unpack_if_needed(batch[schema[Columns.NEXT_OBS]][i]),
+                        convert(unpack_if_needed(obs), observation_space)
+                        if Columns.OBS in input_compress_columns
+                        else convert(obs, observation_space),
+                        convert(
+                            unpack_if_needed(batch[schema[Columns.NEXT_OBS]][i]),
+                            observation_space,
+                        )
+                        if Columns.NEXT_OBS in input_compress_columns
+                        else convert(
+                            batch[schema[Columns.NEXT_OBS]][i], observation_space
+                        ),
                     ],
                     infos=[
                         {},
                         batch[schema[Columns.INFOS]][i]
                         if schema[Columns.INFOS] in batch
                         else {},
                     ],
-                    actions=[batch[schema[Columns.ACTIONS]][i]],
```

Comment:
Unrelated question on the above:
- Why is the first info always `{}`?
- I think `else {},` would yield a tuple, correct? So the resulting final list would be: `[{}, ({},)]`. Maybe I'm wrong, but can you check this?

Reply:
Great question, because there is only a single INFO column in all the offline data. As we need two in the `Episode` we need to fill in a default and this is filled in at timestep zero.

YOur second point is valid - this could lead to a tuple, even though the comma should be part of the list ... I check this. Thanks1

Reply:
Ah, yeah, makes sense. We don't have a `NEXT_INFOS`.

Hmm, I wonder whether this could be a general problem in some strange use cases :)

Reply:
I was also thinking about it, but could not came of with cases that need an `info` at ts=0.